internal/pkg/config/config.go

```diff
@@ -284,27 +284,59 @@ func (fm funcMatcher) MatchFunction(path, receiver, name string) bool {
 	return fm.Package.MatchString(path) && fm.Receiver.MatchString(receiver) && fm.Method.MatchString(name)
 }
 
-var readFileOnce sync.Once
-var readConfigCached *Config
-var readConfigCachedErr error
+// readConfigResults reduces disk access across multiple ReadConfig calls.
```

Comment:
Nit: `readConfigResults` is fine, but I think `cachedConfig` may read slightly better.

Reply:
I actually had something similar first, but didn't like the near-collision with `type configCache`.  It's also not just the config anymore, but the config, error, and sync.  I could go with the more verbose `configCacheElement` I suppose?